util/Assert.c

```diff
@@ -29,6 +33,25 @@ void Assert_failure(const char* format, ...)
     va_start(args, format);
     vfprintf(stderr, format, args);
     fflush(stderr);
+
+#ifdef __GLIBC__
+    void *array[20];
+    size_t size;
+    char **strings;
+    size_t i;
+
+    size = backtrace(array, 20);
+    strings = backtrace_symbols(array, size);
+
+    fprintf(stderr, "Backtrace (%zd frames):\n", size);
+    for (i = 0; i < size; i++)
```

Comment:
I prefer c99 variable declaration, IE: `size_t size = backtrace()` and `for (size_t i = 0; i < size; i++)` because it clarifies the meaning of the variable and (in the case of the for loop) reduces it's scope which can prevent real bugs.